### DISCUSSION TITLE
How to force filters?

### DISCUSSION THREAD

**Discussion by ghostlexly** [directus/directus#11865] [Archive]
How can i force filters on all collections ?
Can't find a way on hooks or to create a middleware..

**Comment by rijkvanzanten** [comment]
Have you tried using permissions?

**Reply by ghostlexly** [reply]
How ?
The permissions are only for roles and crud operations.
I need to implement, per example, the meta: * filter to all requests.

**Reply by rijkvanzanten** [reply]
`meta: *` filter? Do you mean setting default query parameters for every request?

**Reply by ghostlexly** [reply]
Yes, the default query parameters ^^

Nvm, i found a way, made a hook:
`
  init("middlewares.before", ({ app }) => {
    app.use((req, res, next) => {
      req.query = { ...req.query, meta: "*" };
      return next();
    });
  });
`
